src/main/java/com/redhat/rhjmc/containerjfr/core/templates/RemoteTemplateService.java

```diff
@@ -0,0 +1,98 @@
+package com.redhat.rhjmc.containerjfr.core.templates;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.openjdk.jmc.common.unit.IConstrainedMap;
+import org.openjdk.jmc.flightrecorder.configuration.events.EventOptionID;
+import org.openjdk.jmc.flightrecorder.controlpanel.ui.configuration.model.xml.XMLModel;
+import org.openjdk.jmc.flightrecorder.controlpanel.ui.configuration.model.xml.XMLTagInstance;
+import org.openjdk.jmc.flightrecorder.controlpanel.ui.model.EventConfiguration;
+
+import com.redhat.rhjmc.containerjfr.core.FlightRecorderException;
+import com.redhat.rhjmc.containerjfr.core.log.Logger;
+import com.redhat.rhjmc.containerjfr.core.net.JFRConnection;
+
+public class RemoteTemplateService implements TemplateService {
+
+    private final JFRConnection conn;
+
+    public RemoteTemplateService(JFRConnection conn) {
+        this.conn = conn;
+    }
+
+    @Override
+    public List<Template> getTemplates() throws FlightRecorderException {
+        try {
+            return getRemoteTemplateModels().stream()
+                    .map(xml -> xml.getRoot())
+                    .map(
+                            root ->
+                                    new Template(
+                                            getAttributeValue(root, "label"),
+                                            getAttributeValue(root, "description"),
+                                            getAttributeValue(root, "provider")))
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
+            throw new FlightRecorderException(e);
+        }
+    }
+
+    @Override
+    public IConstrainedMap<EventOptionID> getEventsByTemplateName(String templateName)
+            throws FlightRecorderException {
+        try {
+            XMLModel model =
+                    getRemoteTemplateModels().stream()
+                            .filter(
+                                    m ->
+                                            m.getRoot().getAttributeInstances().stream()
+                                                    .anyMatch(
+                                                            attr ->
+                                                                    attr.getAttribute()
+                                                                                    .getName()
+                                                                                    .equals("label")
+                                                                            && attr.getValue()
+                                                                                    .equals(
+                                                                                            templateName)))
```

Comment:
Not a review comment, just curious. Is spotbugs applied here? There's a lot of indentation here heh...

Reply:
Yea, that's autoformatted by spotless. Streaming collections and using lambdas like this tends to get a bit ugly because of those formatting rules.

Reply:
Cool, no worries